### ISSUE TITLE
API call throws an error

### ISSUE THREAD

**Issue by endomorphosis** [gradio-app/gradio#2175]
### Describe the bug

Traceback (most recent call last):
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/gradio/routes.py", line 259, in run_predict
    output = await app.blocks.process_api(
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/gradio/blocks.py", line 687, in process_api
    predictions, duration = await self.call_function(fn_index, inputs)
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/gradio/blocks.py", line 604, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
TypeError: 'NoneType' object is not callable

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

https://github.com/endomorphosis/stable-diffusion

### Logs

```shell
Traceback (most recent call last):
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/gradio/routes.py", line 259, in run_predict
    output = await app.blocks.process_api(
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/gradio/blocks.py", line 687, in process_api
    predictions, duration = await self.call_function(fn_index, inputs)
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/gradio/blocks.py", line 604, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
TypeError: 'NoneType' object is not callable
```

### System Info

```shell
Traceback (most recent call last):
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/gradio/routes.py", line 259, in run_predict
    output = await app.blocks.process_api(
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/gradio/blocks.py", line 687, in process_api
    predictions, duration = await self.call_function(fn_index, inputs)
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/gradio/blocks.py", line 604, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/home/barberb/anaconda3/envs/ldo/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
TypeError: 'NoneType' object is not callable
```

### Severity

annoying

**Comment by endomorphosis** [comment]
the API doesn't work, I think because of something with "fn_index" not working properly, because the error doesn't appear when I explicitly include "fn_index" in the code.

**Comment by freddyaboulton** [comment]
Hi @endomorphosis ! Thanks for filing. The big with fn_index in the API was fixed in #2151 but it hasn’t been released yet.

In the meantime, if you want to use gradio programmatically, a possible workaround in the meantime is to load the app and use it like a python function. The guide for that is here: https://gradio.app/using_blocks_like_functions/

**Comment by BannyLon** [comment]
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/gradio/routes.py", line 488, in run_predict
    output = await app.get_blocks().process_api(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/gradio/blocks.py", line 1434, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/gradio/blocks.py", line 1335, in postprocess_data
    prediction_value = block.postprocess(prediction_value)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/gradio/components/gallery.py", line 205, in postprocess
    raise ValueError(f"Cannot process type as image: {type(img)}")
ValueError: Cannot process type as image: <class 'NoneType'>

我的电脑是mac m2。出现这样报错
